server.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/go-chi/chi/v5"
+	"github.com/invopop/jsonschema"
+)
+
+type Server struct {
+	serverConfig *ServerConfig
+	router       *Router
+}
+
+type ServerConfig struct {
+	name string
+	port int
+}
+
+type RegistrationFunctionSchema struct {
+	InputSchema  *jsonschema.Schema `json:"input"`
+	OutputSchema *jsonschema.Schema `json:"output"`
+	FnName       string             `json:"functionName"`
+	EndPoint     string             `json:"endpoint"`
+}
+
+// Creates and returns an empty Server, with a basic serverConfig
+func Create(name string) *Server {
+
+	return &Server{
+		serverConfig: &ServerConfig{
+			name: name,
+			port: 3000,
+		},
+		router: &Router{
+			routes: []*Route{},
+		},
+	}
+}
+
+func (s *Server) Router() *Router {
+	return s.router
+}
+
+func (server *Server) Start() {
+	// todo: check if router was nil
+	// todo: check if the routes was empty
+	// todo: any additional checks needed ?
+	// todo: check if the server was already started ?
+
+	// create a new internal go-chi router
+	chi_router := chi.NewRouter()
+
+	// read all the routes from the router
+	routes := server.router.routes
+	for _, route := range routes {
+		newFunc := func(w http.ResponseWriter, r *http.Request) {
+			// Decode the JSON request body into the struct instance
+			structPtr := reflect.New(route.handler.inpType).Interface()
+			decoder := json.NewDecoder(r.Body)
+			decoder.DisallowUnknownFields()
```

Comment:
This isn't helpful. Use this library instead: https://github.com/xeipuuv/gojsonschema. Note: JSON schema should be pre-calculated in order to use this library

Reply:
Can you please explain why is this not helpful ?

Reply:
Json schema includes additional things like:
1. Min and max length of string
2. Enumerations

This type of validation won't be captured with the code currently written.